.circleci/config.yml

```diff
@@ -419,12 +487,37 @@ workflows:
   commit-workflow:
     when:
       and:
-        - not: << pipeline.parameters.publish >>
         - not: << pipeline.parameters.create-release >>
+        - not: << pipeline.parameters.publish >>
+        - not: << pipeline.parameters.pre-publish >>
+    jobs:
+      - build-all
+      - unit-tests
+      - integration-tests
+
+  pre-publish-workflow:
```

Comment:
For now this is going to be its own workflow separate from publish. I'd like to see if we run into any conflicts going forward prior to publishing. Right now this is very happy path.